### PR TITLE
Fix CMake 3.31 warnings

### PR DESCRIPTION
**Pull request by rouault**

### CODE REVIEW

File: python/CMakeLists.txt [hunk 1]
```diff
@@ -406,7 +406,7 @@ foreach(pyfile ${PY_FILES})
     COMMAND ${CMAKE_COMMAND} -E make_directory "${QGIS_PYTHON_OUTPUT_DIRECTORY}"
     COMMAND ${CMAKE_COMMAND} -E copy ${pyfile} "${QGIS_PYTHON_OUTPUT_DIRECTORY}"
     WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
-    DEPENDS ${pyfile}
+    # DEPENDS ${pyfile}
```
Comment (nyalldawson) [inline python/CMakeLists.txt:409]: What's the story here? Does newer cmake auto-determine this?

Comment (rouault) [reply python/CMakeLists.txt:409]: no, that has never been valid and is ignored: cf https://cmake.org/cmake/help/latest/command/add_custom_command.html#build-events
We'd need using some other CMake syntax to handle dependencies

Comment (nyalldawson) [reply python/CMakeLists.txt:409]: Looks like it was there up till 3.31: https://cmake.org/cmake/help/v3.31/command/add_custom_command.html#depends

Comment (nyalldawson) [reply python/CMakeLists.txt:409]: Ok, thanks for the explanation! If this was never working then let's just remove those lines instead of commenting?

Comment (rouault) [reply python/CMakeLists.txt:409]: yes I had an hesitation if we'd want to keep them as a memory that we may want to do something more involved to track dependencies. But I'm fine removing them too

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**